example_client/face_detection.py

```diff
@@ -0,0 +1,131 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import grpc
+import numpy as np
+import os
+import tensorflow.contrib.util as tf_contrib_util
+import classes
+import datetime
+import argparse
+from tensorflow_serving.apis import predict_pb2
+from tensorflow_serving.apis import prediction_service_pb2_grpc
+from client_utils import print_statistics
+import cv2
+
+
+parser = argparse.ArgumentParser(description='Demo for face detection requests via TFS gRPC API.'
+                                             'analyses input images and saves with with detected faces.'
+                                             'it relies on model face_detection...')
+
+parser.add_argument('--input_images_dir', required=False, help='Directory with input images', default="images/people")
+parser.add_argument('--output_dir', required=False, help='Directory for staring images with detection results', default="results")
+parser.add_argument('--batch_size', required=False, help='how many images should be grouped in one batch', default=1, type=int)
+parser.add_argument('--width', required=False, help='how many images should be grouped in one batch', default=1200, type=int)
+parser.add_argument('--height', required=False, help='how many images should be grouped in one batch', default=800, type=int)
+parser.add_argument('--grpc_address',required=False, default='localhost',  help='Specify url to grpc service. default:localhost')
+parser.add_argument('--grpc_port',required=False, default=9000, help='Specify port to grpc service. default: 9000')
+
+args = vars(parser.parse_args())
+
+channel = grpc.insecure_channel("{}:{}".format(args['grpc_address'],args['grpc_port']))
+stub = prediction_service_pb2_grpc.PredictionServiceStub(channel)
+
+files = os.listdir(args['input_images_dir'])
+batch_size = args['batch_size']
+print(files)
+print(type(files))
+
+def load_image(file):
+    with open(file, mode='rb') as file:
+        content = file.read()
+
+    img = np.frombuffer(content, dtype=np.uint8)
+    img = cv2.imdecode(img, cv2.IMREAD_COLOR)  # BGR format
+    img = img.astype('float32')
+    img = cv2.resize(img, (args['width'], args['height']))
+    img = img.astype('float32')
+    print(img.shape)
+    cv2.imwrite('test.jpg',img)
+    img = img.transpose(2,0,1).reshape(1,3,args['height'],args['width'])
+    print(img.shape)
+
+    return img
+
+imgs = np.zeros((0,3,args['height'],args['width']), np.dtype('<f'))
+for i in files:
+    img = load_image(os.path.join(args['input_images_dir'], i))
+    imgs = np.append(imgs, img, axis=0)
+
+print(imgs.shape)
+
+iterations = len(files)//batch_size
+
+print('Start processing:')
+print('Iterations: {} \n'.format(iterations))
+
+iteration = 0
+processing_times = np.zeros((0),int)
+
+while iteration <= iterations:
+    for x in range(0, imgs.shape[0] - batch_size + 1, batch_size):
+        iteration += 1
+        if iteration > iterations: break
```

Comment:
Is it necessary? 
Wouldn't changing `iteration <= iterations` to` iteration < iterations` in while loop condition do the job?